### PR TITLE
fix: clarify the wording on stateless actions

### LINKED ISSUE #252: Clarify stateless actions and installation names
The spec at https://github.com/deislabs/cnab-spec/blob/master/101-bundle-json.md#custom-actions says: "Stateless actions can be invoked on a non-existing installation."

What is the intent of this?

1. It is legal to perform a stateless action with an installation name that does not correspond to any installation, or without an installation.

2. It is legal to perform a stateless action without an installation (but if you do give a name then it has to exist).

### PR DESCRIPTION
**Pull request by technosophos**
Closes #252 

Signed-off-by: Matt Butcher <matt.butcher@microsoft.com>

### CODE REVIEW

File: 101-bundle-json.md [hunk 1]
```diff
@@ -769,8 +769,9 @@ Each action is accompanied by a description, which contains the following fields
 
 The `modifies` field MUST be set to `true` if any resource that is managed by the bundle is changed in any way. The `modifies` field assists CNAB implementations in tracking history of changes over time. An implementation of CNAB MAY use this information when describing history or managing releases.
 
-The `stateless` field indicates that the runtime bypass credentials validation (the user MAY or MAY NOT pass any credentials), and will not keep track of this action. Primary intent is to allow invocation bundles to provide dry-run or detailed help functionalities.
-Stateless actions can be invoked on a non-existing installation.
+The `stateless` field indicates that the runtime bypass credentials validation (the user MAY or MAY NOT pass any credentials), and will not keep track of this action. If credentials are supplied, the runtime SHOULD pass the credentials to the invocation image (as in non-stateless actions). An invocation image MAY exit with an error if the necessary credentials are not supplied. However, the runtime MUST NOT fail to invoke the invocation image because of missing credentials.
```
Comment (youreddy) [inline 101-bundle-json.md]: the wording is off on this line: "The `stateless` field indicates that the runtime bypass credentials validation" . How about "The `stateless` field indicates that the runtime will bypass credential validation"

Comment (youreddy) [reply 101-bundle-json.md]: consider changing "non-stateless actions" to "stateful actions"

Comment (technosophos) [reply 101-bundle-json.md]: Made both changes. Thanks.

**Comment by technosophos** [comment]
This just needs a green check mark so I can merge it. :-)

**Comment by radu-matei** [comment]
I think this can be merged now?

**Pull request merged**